src/Model/Post/Media.php

```diff
@@ -269,19 +281,22 @@ private static function addPreviewData(array $media): array
 	 */
 	private static function addActivity(array $media): array
 	{
-		$id = Item::fetchByLink($media['url'], 0, ActivityPub\Receiver::COMPLETION_ASYNC);
+		$id = Item::fetchByLink($media['url'], 0, ActivityPub\Receiver::COMPLETION_ASYNC, $media['mimetype'] ?? '');
 		if (empty($id)) {
+			$media['type'] = $media['type'] != self::ACTIVITY ? $media['type'] : self::JSON;
```

Comment:
I prefer a positive match, it seems less confusing to me.
```suggestion
			$media['type'] = $media['type'] == self::ACTIVITY ? self::JSON : $media['type'];
```